x-pack/plugins/monitoring/public/components/elasticsearch/shard_activity/shard_activity.js

```diff
@@ -57,28 +94,50 @@ const ToggleCompletedSwitch = ({ toggleHistory, showHistory }) => (
     <KuiToolBarText>
       <EuiSwitch
         id="monitoring_completed_recoveries"
-        label="Completed recoveries"
+        label={(
+          <FormattedMessage
+            id="xpack.monitoring.elasticsearch.shardActivity.completedRecoveriesLabel"
+            defaultMessage="Completed recoveries"
+          />
+        )}
         onChange={toggleHistory}
         checked={showHistory}
       />
     </KuiToolBarText>
   </KuiToolBarSection>
 );
 
-export class ShardActivity extends React.Component {
+class ShardActivityUI extends React.Component {
   constructor(props) {
     super(props);
     this.getNoDataMessage = this.getNoDataMessage.bind(this);
   }
 
   getNoDataMessage() {
     if (this.props.showShardActivityHistory) {
-      return 'There are no historical shard activity records for the selected time range.';
+      return this.props.intl.formatMessage({
+        id: 'xpack.monitoring.elasticsearch.shardActivity.noDataMessage',
+        defaultMessage: 'There are no historical shard activity records for the selected time range.'
+      });
     }
     return (
       <Fragment>
-        There are no active shard recoveries for this cluster.<br />
-        Try viewing <EuiLink onClick={this.props.toggleShardActivityHistory}>completed recoveries</EuiLink>.
+        <FormattedMessage
+          id="xpack.monitoring.elasticsearch.shardActivity.noActiveShardRecoveriesMessage"
+          defaultMessage="There are no active shard recoveries for this cluster.{nextLine}
+          Try viewing {shardActivityHistoryLink}."
+          values={{
+            shardActivityHistoryLink: (
+              <EuiLink onClick={this.props.toggleShardActivityHistory}>
+                <FormattedMessage
+                  id="xpack.monitoring.elasticsearch.shardActivity.noActiveShardRecoveriesMessage.completedRecoveriesLinkText"
+                  defaultMessage="completed recoveries"
+                />
+              </EuiLink>
+            ),
+            nextLine: (<br />)
```

Comment:
Just curious, why not just separate this into two separate messages and putting the `<br/>` in between? The first message makes sense as `xpack.monitoring.elasticsearch.shardActivity.noActiveShardRecoveriesMessage.completedRecoveriesLinkTextProblem` and the second one could be `xpack.monitoring.elasticsearch.shardActivity.noActiveShardRecoveriesMessage.completedRecoveriesLinkTextSolution` or something?